### PR TITLE
also fuzz exif makernotes

### PR DESCRIPTION
**Pull request by msmeissn**
This will fuzz EXIF MakerNotes, a huge sub-tree of EXIF data.

### CODE REVIEW

**Comment by msmeissn** [comment]
> @googleb

@googlebot I signed it!

**Comment by inferno-chromium** [comment]
@reaperhulk - can you review this and are you ok with contact addition.

File: projects/libexif/exif_loader_fuzzer.cc [hunk 1]
```diff
@@ -12,6 +12,35 @@ void data_func(ExifContent *content, void *user_data) {
...
+                return 1;
+
+        exif_mnote_data_ref (md);
+        exif_mnote_data_unref (md);
+
+        c = exif_mnote_data_count (md);
+        for (i = 0; i < c; i++) {
+                const char *name = exif_mnote_data_get_name (md, i);
+                if (!name) break;
+                sprintf (buf, "  Name: '%s'\n", exif_mnote_data_get_name (md, i));
```
Comment (reaperhulk) [inline projects/libexif/exif_loader_fuzzer.cc]: These buffers can overflow, but do we need to call sprintf here? What does it get us beyond just calling the function and ignoring the return?

File: projects/libexif/exif_loader_fuzzer.cc [hunk 2]
```diff
@@ -12,6 +12,35 @@ void data_func(ExifContent *content, void *user_data) {
...
+        if (!md)
+                return 1;
+
+        exif_mnote_data_ref (md);
+        exif_mnote_data_unref (md);
+
+        c = exif_mnote_data_count (md);
+        for (i = 0; i < c; i++) {
+                const char *name = exif_mnote_data_get_name (md, i);
+                if (!name) break;
```
Comment (reaperhulk) [inline projects/libexif/exif_loader_fuzzer.cc]: Could we use braces everywhere please? We should also expand the corpus to contain  samples that will trigger these paths if we don’t already have them.

Comment (msmeissn) [reply projects/libexif/exif_loader_fuzzer.cc]: I added braces on these.

Not sure where the corpus is kept, the referenced samples https://github.com/ianare/exif-samples/tree/master/jpg contain maker notes at least.

File: projects/libexif/exif_loader_fuzzer.cc (hunk 1 above)
Comment (msmeissn) [reply projects/libexif/exif_loader_fuzzer.cc]: I removed the unnecessary sprintfs and buffer usage.

**Comment by Dor1s** [comment]
Is this ready to merge?

**Comment by reaperhulk** [comment]
I haven't heard anything from @dfandrich so I'm not sure we can add a new person to the CC list. I'm okay with it, but I'm not the owner of libexif 😞

**Comment by dfandrich** [comment]
Sorry, I'm here but haven't had a chance to look at this yet. I'll try to do so
in the next day or two.

**Comment by msmeissn** [comment]
@dfandrich ? my email additiion needs your approval ;)

(if not, I can remove myself again and just send the code)

**Comment by dfandrich** [comment]
I'm fine with @msmeissn being added to the project.

**Pull request merged**